lib/AST/DiagnosticConsumer.cpp

```diff
@@ -22,7 +22,72 @@
 #include "llvm/Support/raw_ostream.h"
 using namespace swift;
 
-DiagnosticConsumer::~DiagnosticConsumer() { }
+DiagnosticConsumer::~DiagnosticConsumer() = default;
+
+llvm::SMLoc DiagnosticConsumer::getRawLoc(SourceLoc loc) {
+  return loc.Value;
+}
+
+FileSpecificDiagnosticConsumer::FileSpecificDiagnosticConsumer(
+    MutableArrayRef<ConsumerPair> consumers) {
+  for (ConsumerPair &pair : consumers) {
```

Comment:
My style would be to extract out the body of this constructor into a function called something like `buildSortedConsumerMap`. That function would in turn call `populateMap` then `sortMap`. Even though the code is clear as is, I like to make it even easier to read quickly.

Reply:
I don't think this one pulls its weight. I added comments, though.